ydb/docs/en/core/reference/ydb-sdk/topic.md

```diff
@@ -1319,12 +1459,126 @@ When reading starts, the client code must transmit the starting consumer offset
 
 - Java
 
-  Setting the starting offset for reading is not supported in the current state of Java SDK.
+  Starting offset for reading in Java can be set only for AsyncReader.
+  In `StartPartitionSessionEvent` callback `StartPartitionSessionSettings` with desired ReadOffset can be passed to `confirm` method.
+  Offset that should be considered as committed can be set with `setCommittedOffset`.
+
+  ```java
+  @Override
+  public void onStartPartitionSession(StartPartitionSessionEvent event) {
+      event.confirm(StartPartitionSessionSettings.newBuilder()
+              .setReadOffset(lastReadOffset)
+              .setCommitOffset(lastCommitOffset)
+              .build());
+  }
+  ```
 
   The `setReadFrom` setting is used for reading only messages with write timestamps no less than the given one.
 
 {% endlist %}
 
+### Reading without a Consumer {#no-consumer}
+Usually reading progress is saved on server within each Consumer. Though such progres can not be saved if a reader is created without a `Consumer`
+
+{% list tabs %}
+
+- Java
+
+  To read without a Consumer, `withoutConsumer()` method should be called explicitly on `ReaderSettings` Builder:
+
+  ```java
+  ReaderSettings settings = ReaderSettings.newBuilder()
+          .withoutConsumer()
+          .addTopic(TopicReadSettings.newBuilder()
+                  .setPath(TOPIC_NAME)
+                  .build())
+          .build();
+  ```
+  In this case reading progress on server will be lost on PartitionSession restart.
+  To avoid reading from the beginning each time, starting offsets should be set each time on `PartitionSession` start:
+
+  ```java
+  @Override
+  public void onStartPartitionSession(StartPartitionSessionEvent event) {
+      event.confirm(StartPartitionSessionSettings.newBuilder()
+              .setReadOffset(lastReadOffset) // last offset read by client
+              .build());
+  }
+  ```
+
+{% endlist %}
+
+### Reading in transaction {#read-tx}
+
+{% list tabs %}
+
+- Java (sync)
+
+  [Example on GitHub](https://github.com/ydb-platform/ydb-java-examples/blob/develop/ydb-cookbook/src/main/java/tech/ydb/examples/topic/transactions/TransactionReadSync.java)
+
+  Transaction can be set in `ReceiveSettings` for `receive` method:
+
+  ```java
+  Message message = reader.receive(ReceiveSettings.newBuilder()
+          .setTransaction(transaction)
+          .build());
+  ```
+  Such received message will be committed with this transaction. And shouldn't be committed directly.
+  `receive` method sends `sendUpdateOffsetsInTransaction` request on server to link message offset with this transaction and blocks until response is received.
+
+  Transaction requirements:
```

Comment:
Given the same requirements for multiple features/tabs blocks, it's better to put them into an include file to avoid copy-paste.

Reply:
example of an include: https://github.com/ydb-platform/ydb/blob/main/ydb/docs/en/core/operations/query_plans_optimization.md?plain=1